lib/Backends/NNPI/tests/NNPIParameterSweepTest.cpp

```diff
@@ -18,3 +18,91 @@
 using namespace glow;
 
 std::set<std::string> glow::backendTestBlacklist = {};
+
+struct SoftwareOnlyTests {
```

Comment:
The design here is a bit strange. It looks like we're setting up a blacklist for ICEREF here, but then later on in `EmulatorOnlyTests` we're overriding that with different tests?

Is ICEREF's blacklist not a subset of HW's blacklist? If yes, why isn't it a subset?

If no, the design here seems kind of weird/misleading, would prefer to check `USE_INF_API` in here to make things more clear. And I'd also like to cleanup the names of the structs because they're confusing/misleading.